### PR TITLE
[TS] migrate onchain_reading.js

### PR DESCRIPTION
**Pull request by fleupold**
☝️ 

Nothing special here. Again adding type annotations (typing the async generator was a bit tricky) and adjusting all call sites.

Turns out that the block number is actually not a standard argument for the `.call` method. As a workaround I'm creating a copy of the contract and setting the default block before invoking the method

### Test Plan

CI

### CODE REVIEW

File: src/onchain_reading.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+} from "./encoding";
+import { BatchExchangeViewer } from "../build/types/BatchExchangeViewer";
+import { BatchExchange } from "../build/types/BatchExchange";
+import BN from "bn.js";
+
+/**
+ * Returns an iterator yielding an item for each page of order in the orderbook that is currently being collected.
+ * @param {BatchExchangeViewer} contract to query from
+ * @param {number} pageSize the number of items to fetch per page
+ * @param {number?} blockNumber the block number to execute the query at, defaults to "latest" if omitted
```
Comment (e00E) [inline src/onchain_reading.ts]: Nit: Does this style of comment still make sense for typescript? The type of the parameter is already in the signature.

File: src/onchain_reading.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  }
+};
+
+/**
+ * Returns open orders in the orderbook.
+ * @param {BatchExchangeViewer} contract to query from
+ * @param {number} pageSize the number of items to fetch per page
+ * @param {number?} blockNumber the block number to execute the query at, defaults to "latest" if omitted
+ */
+export const getOpenOrders = async (
```
Comment (e00E) [inline src/onchain_reading.ts]: Why does this not use the `export function foo` syntax and instead uses `export const foo = ...`?

**Review by e00E — APPROVED** [review]

File: src/onchain_reading.ts (hunk 2 above)
Comment (nlordell) [reply src/onchain_reading.ts]: I would (also?) prefer the `export function foo` syntax here.

**Review by nlordell — APPROVED** [review]

File: src/onchain_reading.ts (hunk 1 above)
Comment (bh2smith) [reply src/onchain_reading.ts]: Agree.

File: src/onchain_reading.ts (hunk 2 above)
Comment (bh2smith) [reply src/onchain_reading.ts]: I, on the other hand, simply look forward to seeing what `export function foo` winds up looking like.

File: src/onchain_reading.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  }
+
+  while (hasNextPage) {
+    const page = await contract.methods
+      .getOpenOrderBookPaginated([], nextPageUser, nextPageUserOffset, pageSize)
+      .call();
+    const elements = decodeIndexedOrders(page.elements);
+    yield elements;
+
+    //Update page info
```
Comment (bh2smith) [inline src/onchain_reading.ts:38]: ubernit:
```suggestion
    // Update page info
```

**Review by bh2smith — APPROVED** [review]

File: src/onchain_reading.ts (hunk 2 above)
Comment (fleupold) [reply src/onchain_reading.ts]: Sure. There was no specific reason, I just applied the minimum number of keystrokes to get it to compile. Happy to remove the const.

**Pull request merged**